checker.cpp

```diff
@@ -1,46 +1,60 @@
 #include <iostream>
-#include <assert.h>
+#include <cassert>
 #include <unistd.h>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+void simulateBlinkingEffect() {
+    for (int i = 0; i < 6; i++) {
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
     }
+}
+
+bool isOutOfRange(float value, float lower, float upper) {
+    return value < lower || value > upper;
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+int vitalsOk(float temperature, float pulseRate, float spo2) {
```

Comment:
As discussed, this function can be simplified. It's currently failing the complexity check, indicating that the complexity can keep building here, as we extend functionality.